### PR TITLE
Refactor logs controller

### PR DESCRIPTION
**Pull request by vidoseaver**
## Overview
Refactor logs_controller
Change d to date in #by_day
Change t to total in #stats_service
Change l to log_part, change w to word
Change @l to @log in #destroy
Change @l to location in #destroy
Change l to logs, l to log and lv to log_volunteer in #leave
Change @loc to @location in #take
Change lg to log_part in #parse_and_create_log_part
Change lp to log_part in #finalize_log

### CODE REVIEW

File: app/controllers/logs_controller.rb [hunk 1]
```diff
@@ -95,24 +95,24 @@ def stats_service
...
-          next if w =~ /(nothing|no |none)/ or w =~ /etc/ or w =~ /n\/a/ or w =~ /misc/
+      LogPart.select('description').where('description IS NOT NULL').each{ |log_part|
+        log_part.description.strip.split(/\s*\,\s*/).each{ |word|
+          word = word.strip.downcase.tr(',', '')
+          next if word =~ /(nothing|no |none)/ or word =~ /etc/ or word =~ /n\/a/ or word =~ /misc/
           # people cannot seem to spell the most delicious fruit correctly
-          w = 'avocados' if w == 'avacados' or w == 'avocadoes' or w == 'avocado'
-          words[w] = 0 if words[w].nil?
-          words[w] += 1
+          word = 'avocados' if word == 'avacados' or word == 'avocadoes' or word == 'avocado'
```
Comment (rylanb) [inline app/controllers/logs_controller.rb]: huh.... wonder what this is even doing in here.

File: app/controllers/logs_controller.rb [hunk 2]
```diff
@@ -95,24 +95,24 @@ def stats_service
...
+          words[word] = 0 if words[word].nil?
+          words[wotd] += 1
```
Comment (rylanb) [inline app/controllers/logs_controller.rb]: misspelling here

File: app/controllers/logs_controller.rb [hunk 3]
```diff
@@ -283,19 +283,19 @@ def take
...
+    logs.each { |log| authorize! :leave, log }
 
-    l.each do |x|
-      if x.has_volunteer? current_volunteer
-        LogVolunteer.where(:volunteer_id=>current_volunteer.id, :log_id=>x.id).each{ |lv|
-          lv.active = false
-          lv.save
+    logs.each do |log|
+      if log.has_volunteer? current_volunteer
+        LogVolunteer.where(:volunteer_id=>current_volunteer.id, :log_id=>log.id).each{ |log_volunteer|
```
Comment (rylanb) [inline app/controllers/logs_controller.rb]: Could you update to volunteer_id: current_volunteer.id new hash syntax too?

**Comment by rylanb** [comment]
Nice job here @vidoseaver, just a couple small things! 

Thanks a ton!

File: app/controllers/logs_controller.rb (hunk 1 above)
Comment (ericfreese) [inline app/controllers/logs_controller.rb]: could clean this up a little by: `%w(avacados avocadoes avocado).include?(word)`

Comment (rylanb) [inline app/controllers/logs_controller.rb]: @ericfreese what's the reco-ed way to do || and && changes... does Rubocop have a setting for that?

I think and / or use is not recommended?

Comment (ericfreese) [inline app/controllers/logs_controller.rb]: yeah `and` and `or` are pretty low precedence so act a little differently than `&&` and `||`

http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/

There's probably a rubocop setting for it but not sure what off the top of my head

**Comment by moveson** [comment]
Would that line of code be better if it were:

`word = 'avocados' if %w(avacados avocadoes avocado).include?(word)`

File: app/controllers/logs_controller.rb (hunk 2 above)
Comment (vidoseaver) [inline app/controllers/logs_controller.rb]: ahh! yup

File: app/controllers/logs_controller.rb (hunk 1 above)
Comment (vidoseaver) [inline app/controllers/logs_controller.rb]: ok, i'd never seen or before and I thought about changing it to || but was airing on the side of caution. I'll change it to an array and include?

File: app/controllers/logs_controller.rb (hunk 3 above)
Comment (vidoseaver) [inline app/controllers/logs_controller.rb]: Yup!

**Comment by vidoseaver** [comment]
fixed requested changes.
Change wotd to word line 115 of logs_controller
Change line 113 of logs_controller to use arary instead of or's
Update line 296 to use new hash syntax

**Comment by ericfreese** [comment]
@vidoseaver Can you rebase this on `master`, run rubocop, and fix any offenses that you see. I'm seeing the following after rebasing locally:

```
$ bundle exec rubocop
Inspecting 185 files
.........W...............................................................................................................................................................................

Offenses:

app/controllers/logs_controller.rb:287:11: C: Use 2 (not -1) spaces for indentation.
          [Log.find(params[:id])]
          ^
app/controllers/logs_controller.rb:288:9: C: Align else with unless.
        else
        ^^^^
app/controllers/logs_controller.rb:290:9: W: end at 290, 8 is not aligned with unless at 286, 11.
        end
        ^^^

185 files inspected, 3 offenses detected
```

**Comment by vidoseaver** [comment]
@ericfreese Sorry, I'm not sure what you mean by rebasing on master. 
Here is what i tried to do:
git checkout master
git pull origin master
git checkout refactor-logs-controller
git rebase master 
Got a merge issue with the gemfile.lock and the controller... 

Git reset hard on this refactor branch, merged in the master. Resolved merge conflicts but
when I run bundle exec rubocop i have 193 offences.

How shall I proceed? Also I seem to no longer be in the slack

**Comment by ericfreese** [comment]
@vidoseaver 

>Sorry, I'm not sure what you mean by rebasing on master.
>Here is what i tried to do:
>git checkout master
>git pull origin master
>git checkout refactor-logs-controller
>git rebase master

This is exactly what I meant :)

> Got a merge issue with the gemfile.lock and the controller...

I'm not sure why you would have had a merge issue with the gemfile.lock, but it may have been easiest to resolve the conflicts at this point instead of hard resetting and merging master into your branch.

> when I run bundle exec rubocop i have 193 offences

Is this still the case? It looks like rubocop is passing on Circle CI.

> Also I seem to no longer be in the slack

It looks to me like you're still in there. I sent ya a message. Maybe you'll get a notification or something. I can try to send you another invite if you'd like.

**Comment by rylanb** [comment]
@vidoseaver There are merge conflicts for this branch... can you look into that? Then I can merge. 

Thanks!

**Comment by vidoseaver** [comment]
So I pulled master and merged it but it kept saying up to date and I wasn't getting any conflicts so I tried rebasing again (still not sure what it is)  I'm seeing "this branch has no conflicts with the base branch when rebasing" so I think we're good to go?

**Comment by vidoseaver** [comment]
@ericfreese Yeah, I have 188 offences with rubocop locally. Not sure what the deal is...

File: app/controllers/food_types_controller.rb [hunk 4]
```diff
@@ -14,7 +14,6 @@ def destroy
     authorize! :destroy, food_type
     food_type.active = false
     food_type.save
-    redirect_to(request.referrer)
```
Comment (ericfreese) [inline app/controllers/food_types_controller.rb]: Why are we removing this `redirect_to`?

Comment (vidoseaver) [inline app/controllers/food_types_controller.rb]: We shouldn't be I must of accidently deleted it when handling merge conflicts

**Comment by vidoseaver** [comment]
Now code climate doesn't like it because the destroy methods are too similar... hmmm

**Comment by ericfreese** [comment]
> Now code climate doesn't like it because the destroy methods are too similar

I think we can ignore code climate in this case. We've been seeing this a fair bit, where extracting duplicated code into some shared method would only make the code harder to reason about.

We should probably increase code climate's threshold for flagging duplicate code. From https://docs.codeclimate.com/docs/duplication:

> If the engine is too easily reporting duplication, try raising the threshold. If you suspect that the engine isn't catching enough duplication, try lowering the threshold. The best setting tends to differ from language to language.

**Comment by vidoseaver** [comment]
Is that something you want me to do?

**Comment by ericfreese** [comment]
@vidoseaver Nah, we can handle that some other time.

**Comment by ericfreese** [comment]
This looks good to me. Only thing that I think would be nice would be to maybe squash this down into one commit, but I'm fine with it as-is. @rylanb you want to take another look and merge when/if you're ready?

**Comment by rylanb** [comment]
ran `git rebase -i master` and squashed down to 1 commit.

**Pull request merged**